pkg/controller/global/template/template_init_data.go

```diff
@@ -0,0 +1,273 @@
+		return nil, err
+	}
+
+	return bytes.Split(templateBuffer.Bytes(), []byte("\n---\n")), nil
+}
+
+func initBaseTemplate(vmTemplates ctlapisv1alpha1.VirtualMachineTemplateClient) error {
+	ymls, err := generateYmls(templateTmpl)
+	if err != nil {
+		return err
+	}
+
+	for _, yml := range ymls {
+		var vmTemplate apisv1alpha1.VirtualMachineTemplate
+		if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(yml), 1024).Decode(&vmTemplate); err != nil {
+			return errors.Wrap(err, "Failed to convert virtualMachineTemplate from yaml to object")
+		}
+
+		if _, err := vmTemplates.Create(&vmTemplate); err != nil && !apierrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "Failed to create virtualMachineTemplate %s:%s", vmTemplate.Namespace, vmTemplate.Name)
+		}
+	}
+	return nil
+}
+
+func initBaseTemplateVersion(vmTemplateVersions ctlapisv1alpha1.VirtualMachineTemplateVersionClient) error {
+	ymls, err := generateYmls(templateVersionTmpl)
+	if err != nil {
+		return err
+	}
+
+	for _, yml := range ymls {
+		var vmTemplateVersion apisv1alpha1.VirtualMachineTemplateVersion
+		if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(yml), 1024).Decode(&vmTemplateVersion); err != nil {
+			return errors.Wrap(err, "Failed to convert virtualMachineTemplateVersion from yaml to object")
+		}
+
+		if _, err := vmTemplateVersions.Create(&vmTemplateVersion); err != nil && !apierrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "Failed to create virtualMachineTemplateVersion %s:%s", vmTemplateVersion.Namespace, vmTemplateVersion.Name)
+		}
+	}
+	return nil
+}
+
+var (
+	initBaseTemplates = `
+apiVersion: harvester.cattle.io/v1alpha1
+kind: VirtualMachineTemplate
+metadata:
+  name: iso-image-base-template
+  namespace: {{ .Namespace }}
+spec:
+  description: Template for booting virtual machine from an ISO image
+---
+apiVersion: harvester.cattle.io/v1alpha1
+kind: VirtualMachineTemplate
+metadata:
+  name: raw-image-base-template
+  namespace: {{ .Namespace }}
+spec:
+  description: Template for booting virtual machine from a qcow2/raw image
+---
+apiVersion: harvester.cattle.io/v1alpha1
+kind: VirtualMachineTemplate
+metadata:
+  name: windows-iso-image-base-template
+  namespace: {{ .Namespace }}
+spec:
+  description: Template for booting Windows virtual machine from an ISO image
+`
+
+	initBaseTemplateVersions = `
+apiVersion: harvester.cattle.io/v1alpha1
+kind: VirtualMachineTemplateVersion
+metadata:
+  name: iso-image-base-version
+  namespace: {{ .Namespace }}
+spec:
+  templateId: {{ .Namespace }}:iso-image-base-template
+  vm:
+    running: true
+    template:
+      spec:
+        domain:
+          cpu:
+            cores: 1
+          devices:
+            disks:
+            - disk:
+                bus: virtio
+              name: rootdisk
+              bootOrder: 1
+            - cdrom:
+                bus: sata
+                readonly: true
+              name: cdrom-disk
+              bootOrder: 2
```

Comment:
can we please add the minimum pod network configuration and this is required NIC of each VM when creating from the UI.
refer to https://kubevirt.io/user-guide/#/creation/interfaces-and-networks?id=pod

Reply:
the pod network will be added to VMI by default event we don't define it, but I will add it, so UI can show it.